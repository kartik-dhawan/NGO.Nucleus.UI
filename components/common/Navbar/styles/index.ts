export const styles = {
  navbarWrapper: {
    display: "flex",
    margin: 0,
    padding: 0,
    borderBottom: "0.5px solid #999",
    position: "fixed",
    top: 0,
    left: 0,
    right: 0,
    zIndex: 100,
    backgroundColor: "#fff",
  },
  navbarAppTitleWrapper: {
    "& > img": {
      height: "40px",
      width: "40px",
    },
    display: "flex",
    alignItems: "center",
    gap: "4px",
    margin: {
      xs: "0px 8px",
      md: "0px 28px",
    },
    padding: "8px 0px",
  },
  navbarTitle: {
    width: "5rem",
    fontSize: {
      xs: "14px",
      md: "16px",
    },
  },
  navbarItemsWrapper: {
    display: {
      xs: "none",
      md: "flex",
    },
    padding: 0,
    width: "100%",
    "& > a:nth-of-type(5)": {
      marginLeft: "auto",
    },
    "& > a:last-child": {
      backgroundColor: "#fcef48",
    },
  },
  navbarItem: {
    width: "100%",
    padding: "8px 24px",
    "&:hover": {
      backgroundColor: "#fff898",
    },
    color: "#111",
    height: "100%",
    transition: "200ms all ease",
    fontWeight: 700,
    display: "flex",
    alignItems: "center",
    "& > svg": {
      fontSize: "18px",
      margin: {
        xs: "0px 0px 0px auto",
        md: "0px 8px 0px 0px",
      },
    },
  },
  closeMenuButton: {
    textTransform: "capitalize",
    fontSize: "1rem",
    textDecoration: "underline",
    color: "#111",
  },
  mobileDonateButton: {
    backgroundColor: "#fcef48",
    "&:hover": {
      backgroundColor: "#fcef48",
    },
    display: {
      xs: "flex",
      md: "none",
    },
    width: "max-content",
    color: "#111",
    textDecoration: "none",
    padding: "0px 24px",
  },
  openMenuButton: {
    color: "#111",
    textTransform: "capitalize",
    fontWeight: 700,
    fontSize: "1rem",
    padding: "8px 24px",
    display: {
      xs: "flex",
      md: "none",
    },
    marginLeft: "auto",
    "&:hover": {
      backgroundColor: "#fff898",
    },
  },
}
