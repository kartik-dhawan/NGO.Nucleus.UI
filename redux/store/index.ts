import { configureStore } from "@reduxjs/toolkit"
import contentSlice from "../slices/contentSlice"

const store = configureStore({
  reducer: {
    contentSlice,
  },
})

export type RootType = ReturnType<typeof store.getState>

export default store
