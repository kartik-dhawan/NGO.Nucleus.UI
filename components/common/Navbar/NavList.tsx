import { List, ListItem } from "@mui/material"
import { NavbarItem } from "../../../utils/interfaces"
import Link from "next/link"
import LockOutlinedIcon from "@mui/icons-material/LockOutlined"
import { styles } from "./styles"
import { useDispatch, useSelector } from "react-redux"
import { RootType } from "../../../redux/store"
import { toggleLoginDialog } from "../../../redux/slices/authSlice"
import { useEffect, useState } from "react"
import { useRouter } from "next/router"
import { AUTHENTICATED_ADMIN_PAGES } from "../../../utils/constants"

interface NavListProps {
  orientation: "desktop" | "mobile"
  toggleMenu?: any
}

const NavList = ({ orientation, toggleMenu }: NavListProps) => {
  const dispatch = useDispatch()
  const router = useRouter()

  const { content } = useSelector((state: RootType) => state.contentSlice)

  const [isAuthenticated, setIsAuthenticated] = useState<boolean>(false)

  useEffect(() => {
    setIsAuthenticated(localStorage.getItem("isAuthenticated") === "true")
  }, [router])

  const mobileListStyles = {
    ...styles.navbarItemsWrapper,
    display: {
      xs: "flex",
      md: "none",
    },
    flexDirection: "column",
    minWidth: {
      xs: "100vw",
      sm: "20rem",
    },
  }

  return (
    <List
      sx={
        orientation === "desktop" ? styles.navbarItemsWrapper : mobileListStyles
      }
    >
      {content?.navbarItems?.map((item: NavbarItem) => {
        const desktopStyles = {
          ...styles.navbarItem,
          borderLeft: item.id < 4 ? "0.5px solid #c9c9c9" : "none",
          borderRight: item.id > 4 ? "0.5px solid #c9c9c9" : "none",
        }
        const mobileNavItemStyles = {
          ...styles.navbarItem,
          borderBottom: "0.5px solid #222",
          width: "100%",
          display: "flex",
          flexDirection: "row-reverse",
          justifyContent: "flex-end",
          padding: "16px",
        }
        return (
          <Link
            href={
              /**
               * on home page, if user is authenticated, the url will become '/admin'
               * on home page, if user is unauthenticated, the url will become '/#!'
               * on other pages, if user is authenticated, the url will become '/logout'
               */
              item.label === "Admin"
                ? AUTHENTICATED_ADMIN_PAGES.includes(router.asPath)
                  ? "/logout"
                  : isAuthenticated
                  ? "/admin"
                  : "#!"
                : item.url
            }
            key={item.id}
            style={{
              textDecoration: "none",
              width: orientation === "mobile" ? "100%" : "max-content",
            }}
            onClick={() => {
              item.label === "Admin" && // eslint-disable-line
                !isAuthenticated &&
                dispatch(toggleLoginDialog())
              toggleMenu && toggleMenu() // eslint-disable-line
            }}
          >
            <ListItem
              sx={
                orientation === "desktop" ? desktopStyles : mobileNavItemStyles
              }
            >
              {item.id === 4 && !isAuthenticated && <LockOutlinedIcon />}
              {item.label === "Admin" &&
              AUTHENTICATED_ADMIN_PAGES.includes(router.asPath)
                ? "Logout"
                : item.label}
            </ListItem>
          </Link>
        )
      })}
    </List>
  )
}

export default NavList
