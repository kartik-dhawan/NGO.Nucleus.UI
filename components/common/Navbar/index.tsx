import { Box, Button, Drawer, Typography } from "@mui/material"
import { useSelector } from "react-redux"
import { RootType } from "../../../redux/store"
import Image from "next/image"
import Link from "next/link"
import { styles } from "./styles"
import { useCallback, useState } from "react"
import NavList from "./NavList"

const Navbar = () => {
  const [openMenu, setOpenMenu] = useState<boolean>(false)

  const { content } = useSelector((state: RootType) => state.contentSlice)
  const navbarListLength = (content && content.navbarItems?.length) ?? 4

  const toggleMenu = useCallback(() => {
    setOpenMenu((state) => !state)
  }, [])

  return (
    <Box className="robotoCondensed" component="nav" sx={styles.navbarWrapper}>
      <Box sx={styles.navbarAppTitleWrapper}>
        {content?.appIcon && (
          <Image
            src={content?.appIcon?.fields.file.url}
            alt={content?.appIcon?.fields.description}
            height={32}
            width={32}
          />
        )}
        <Typography sx={styles.navbarTitle}>{content.appTitle}</Typography>
      </Box>
      {/* below list only for desktop navbar */}
      <NavList orientation="desktop" />
      {/* below section for mobile navbar */}
      {/* Menu button only visible in portrait-tablets/mobiles */}
      <Button
        disableRipple
        className="robotoCondensed"
        onClick={toggleMenu}
        sx={styles.openMenuButton}
      >
        Menu
      </Button>

      {content.navbarItems && (
        <Link
          href={content.navbarItems[navbarListLength - 1].url}
          style={{ textDecoration: "none" }}
        >
          <Box
            sx={{
              ...styles.navbarItem,
              ...styles.mobileDonateButton,
            }}
          >
            {content.navbarItems[navbarListLength - 1].label}
          </Box>
        </Link>
      )}
      <Drawer
        anchor="right"
        open={openMenu}
        onClose={toggleMenu}
        sx={{
          display: {
            md: "none",
          },
        }}
      >
        <NavList orientation="mobile" />
        <Button
          disableRipple
          onClick={toggleMenu}
          sx={styles.closeMenuButton}
          className="robotoCondensed"
        >
          Close
        </Button>
      </Drawer>
    </Box>
  )
}
export default Navbar
