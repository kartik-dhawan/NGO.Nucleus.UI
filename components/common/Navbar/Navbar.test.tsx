import renderer from "react-test-renderer"
import Navbar from "."
import { Provider } from "react-redux"
import store from "../../../redux/store"
import "@testing-library/jest-dom"

describe("Navbar component", () => {
  it("-- should render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Navbar />
        </Provider>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
