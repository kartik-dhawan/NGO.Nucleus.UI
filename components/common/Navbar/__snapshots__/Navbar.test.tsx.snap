// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component -- should render correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1sajmd3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin:0px;padding:0px;border-bottom:0.5px solid #999;position:-webkit-sticky;position:sticky;top:0;left:0;right:0;z-index:100;background-color:#fff;}",
      }
    }
    data-emotion="css 1sajmd3"
  />,
  <nav
    className="robotoCondensed MuiBox-root css-1sajmd3"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-q2mcz2{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:4px;padding:8px 0px;}@media (min-width:0px){.css-q2mcz2{margin:0px 8px;}}@media (min-width:900px){.css-q2mcz2{margin:0px 28px;}}.css-q2mcz2>img{height:40px;width:40px;}",
        }
      }
      data-emotion="css q2mcz2"
    />
    <div
      className="MuiBox-root css-q2mcz2"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-z829vz-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;width:5rem;}@media (min-width:0px){.css-z829vz-MuiTypography-root{font-size:14px;}}@media (min-width:900px){.css-z829vz-MuiTypography-root{font-size:16px;}}",
          }
        }
        data-emotion="css z829vz-MuiTypography-root"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 css-z829vz-MuiTypography-root"
      />
    </div>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1fpihtw-MuiList-root{list-style:none;margin:0;padding:0;position:relative;padding-top:8px;padding-bottom:8px;padding:0px;width:100%;}@media (min-width:0px){.css-1fpihtw-MuiList-root{display:none;}}@media (min-width:900px){.css-1fpihtw-MuiList-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}}.css-1fpihtw-MuiList-root>a:nth-of-type(5){margin-left:auto;}.css-1fpihtw-MuiList-root>a:last-child{background-color:#fcef48;}",
        }
      }
      data-emotion="css 1fpihtw-MuiList-root"
    />
    <ul
      className="MuiList-root MuiList-padding css-1fpihtw-MuiList-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-bs9nz7-MuiButton-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 8px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;color:#111;text-transform:capitalize;font-weight:700;font-size:1rem;padding:8px 24px;margin-left:auto;}.css-bs9nz7-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);}@media (hover: none){.css-bs9nz7-MuiButton-root:hover{background-color:transparent;}}.css-bs9nz7-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);}@media (min-width:0px){.css-bs9nz7-MuiButton-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}}@media (min-width:900px){.css-bs9nz7-MuiButton-root{display:none;}}.css-bs9nz7-MuiButton-root:hover{background-color:#fff898;}",
        }
      }
      data-emotion="css bs9nz7-MuiButton-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1pm5f6h-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 8px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;color:#111;text-transform:capitalize;font-weight:700;font-size:1rem;padding:8px 24px;margin-left:auto;}.css-1pm5f6h-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-1pm5f6h-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1pm5f6h-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1pm5f6h-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);}@media (hover: none){.css-1pm5f6h-MuiButtonBase-root-MuiButton-root:hover{background-color:transparent;}}.css-1pm5f6h-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);}@media (min-width:0px){.css-1pm5f6h-MuiButtonBase-root-MuiButton-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}}@media (min-width:900px){.css-1pm5f6h-MuiButtonBase-root-MuiButton-root{display:none;}}.css-1pm5f6h-MuiButtonBase-root-MuiButton-root:hover{background-color:#fff898;}",
        }
      }
      data-emotion="css 1pm5f6h-MuiButtonBase-root-MuiButton-root"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium robotoCondensed css-1pm5f6h-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Menu
    </button>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1tz4fy1-MuiDrawer-root{z-index:1200;}@media (min-width:900px){.css-1tz4fy1-MuiDrawer-root{display:none;}}",
        }
      }
      data-emotion="css 1tz4fy1-MuiDrawer-root"
    />
  </nav>,
]
`;
